packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -237,68 +237,77 @@ class DriveCommand extends RunCommandBase {
-            '--no-launch-chrome': true,
-          if (boolArg('multidex'))
-            'multidex': true,
+    try {
+      if (stringArg('use-existing-app') == null) {
+        await driverService.start(
+            buildInfo,
+            device,
+            debuggingOptions,
+            ipv6,
+            applicationBinary: applicationBinary,
+            route: route,
+            userIdentifier: userIdentifier,
+            mainPath: targetFile,
+            platformArgs: <String, Object>{
+              if (traceStartup)
+                'trace-startup': traceStartup,
+              if (web)
+                '--no-launch-chrome': true,
+              if (boolArg('multidex'))
+                'multidex': true,
+            }
+        );
+      } else {
+        final Uri uri = Uri.tryParse(stringArg('use-existing-app'));
+        if (uri == null) {
+          throwToolExit('Invalid VM Service URI: ${stringArg('use-existing-app')}');
         }
-      );
-    } else {
-      final Uri uri = Uri.tryParse(stringArg('use-existing-app'));
-      if (uri == null) {
-        throwToolExit('Invalid VM Service URI: ${stringArg('use-existing-app')}');
+        await driverService.reuseApplication(
+          uri,
+          device,
+          debuggingOptions,
+          ipv6,
+        );
       }
-      await driverService.reuseApplication(
-        uri,
-        device,
-        debuggingOptions,
-        ipv6,
+
+      final int testResult = await driverService.startTest(
+        testFile,
+        stringsArg('test-arguments'),
+        <String, String>{},
+        packageConfig,
+        chromeBinary: stringArg('chrome-binary'),
+        headless: boolArg('headless'),
+        browserDimension: stringArg('browser-dimension').split(','),
+        browserName: stringArg('browser-name'),
+        driverPort: stringArg('driver-port') != null
+          ? int.tryParse(stringArg('driver-port'))
+          : null,
+        androidEmulator: boolArg('android-emulator'),
+        profileMemory: stringArg('profile-memory'),
       );
-    }
+      if (testResult != 0 && screenshot != null) {
+        await _takeScreenshot(device);
+      }
 
-    final int testResult = await driverService.startTest(
-      testFile,
-      stringsArg('test-arguments'),
-      <String, String>{},
-      packageConfig,
-      chromeBinary: stringArg('chrome-binary'),
-      headless: boolArg('headless'),
-      browserDimension: stringArg('browser-dimension').split(','),
-      browserName: stringArg('browser-name'),
-      driverPort: stringArg('driver-port') != null
-        ? int.tryParse(stringArg('driver-port'))
-        : null,
-      androidEmulator: boolArg('android-emulator'),
-      profileMemory: stringArg('profile-memory'),
-    );
-    if (testResult != 0 && screenshot != null) {
-      await takeScreenshot(device, screenshot, _fileSystem, _logger, _fsUtils);
+      if (boolArg('keep-app-running') ?? (argResults['use-existing-app'] != null)) {
```

Comment:
If the code here between 292 and 302 throws, you could get two screenshots, one from the call on 289 and another from the one in the `catch`. Is that intentional?

Reply:
You're right.  Worse, if `testResult` fails which would happen often enough, it will take two screenshots, one from line 289, then another on catching the `toolExit`.  Let me fix that up.

Reply:
Fixed.  Also added a test.

Reply:
Oh right, the screenshot should be taken after the `drive` test failure but before the app is stopped.  Created https://github.com/flutter/flutter/pull/96973 to reinstate this behavior.